### ISSUE TITLE
test.python.visualization.test_visualization.TestLatexSourceGenerator should test against references

### ISSUE THREAD

**Issue by 1ucian0** [Qiskit/qiskit-terra#3364]
The tests in `test.python.visualization.test_visualization.TestLatexSourceGenerator` should check against a file reference.

**Comment by tareqdandachi** [comment]
Does something like this fix the issue? If so I will work on replicating something similar for all the other LatexSource tests. Also, would it be better if I defined the expected content in an external file rather than a variable in every test, or is this fine?

```python
    def test_tiny_circuit(self):
        """Test draw tiny circuit."""
        expected_content = "\\begin{equation*}\\Qcircuit@C=1.0em@R=0.0em@!R{\\lstick{{q}_{0}:\\ket{0}}&\\gate{H}&\\qw&\\qw\\\\}\\end{equation*}\\end{document}"
        ...
        try:
            ...
            with open(filename,'r') as f:
                output = f.read()
                exported_content = re.sub(r'\s+', '', output.split("\\begin{document}")[1])
                self.assertEqual(exported_content, expected_content)
```

**Comment by 1ucian0** [comment]
I think I (kinda accidentally) closed this one with #5062.
